test/tests/DatePickerInput-test.js

```diff
@@ -138,5 +138,55 @@ describe('DatePickerInput', function() {
 
   });
 
+  it('DatePicker should close on select date', function() {
+
+    const inputWrapper = TestUtils.renderIntoDocument(
+      <div className='ui input'>
+        <DatePickerInput onChange={() => {}} showOnInputClick autoClose />
+      </div>
+    );
+
+    let datePickers = TestUtils.scryRenderedComponentsWithType(inputWrapper, DatePicker);
+    expect(datePickers.length).toBe(0);
+
+    const datePickerInputArea = TestUtils.findRenderedDOMComponentWithTag(inputWrapper, 'input');
+    TestUtils.Simulate.click(datePickerInputArea);
+
+    datePickers = TestUtils.scryRenderedComponentsWithType(inputWrapper, DatePicker);
+    expect(datePickers.length).toBe(1);
+
+    const pickerButton = TestUtils.scryRenderedDOMComponentsWithClass(inputWrapper, 'react-datepicker-picker')[0];
+    TestUtils.Simulate.click(pickerButton);
+
+    datePickers = TestUtils.scryRenderedComponentsWithType(inputWrapper, DatePicker);
+    expect(datePickers.length).toBe(0, 'DatePicker didn\'t close correctly');
+
+  });
+
+  // KARMA DOESN'T RETURN keyCode PROPERTY...
+
+  // it('DatePicker should close on enter key event on input', function() {
+
+  //   const inputWrapper = TestUtils.renderIntoDocument(
+  //     <div className='ui input'>
+  //       <DatePickerInput onChange={() => {}} showOnInputClick />
+  //     </div>
+  //   );
+
+  //   let datePickers = TestUtils.scryRenderedComponentsWithType(inputWrapper, DatePicker);
+  //   expect(datePickers.length).toBe(0);
+
+  //   const datePickerInputArea = TestUtils.findRenderedDOMComponentWithTag(inputWrapper, 'input');
+  //   TestUtils.Simulate.click(datePickerInputArea);
+
+  //   datePickers = TestUtils.scryRenderedComponentsWithType(inputWrapper, DatePicker);
+  //   expect(datePickers.length).toBe(1);
+
+  //   TestUtils.Simulate.keyUp(datePickerInputArea, {key: 'Enter'});
```

Comment:
Try: http://stackoverflow.com/questions/30628093/reacts-testutils-simulate-keydown-does-not-work